### DISCUSSION TITLE
v0.2.3

### DISCUSSION THREAD

**Discussion by RobertCraigie** [RobertCraigie/prisma-client-py#85] [General]
## Bug Fixes

- Partial types with enum fields are now correctly generated (#84)

## Prisma Update

The internal Prisma binaries that Prisma Client Python uses have been upgraded from `3.1.1` to `3.3.0`.

- MongoDB introspection support is in preview

For a full list of changes see https://github.com/prisma/prisma/releases/tag/3.2.0 and https://github.com/prisma/prisma/releases/tag/3.3.0

<hr /><em>This discussion was created from the release <a href='https://github.com/RobertCraigie/prisma-client-py/releases/tag/v0.2.3'>v0.2.3</a>.</em>

**Comment by kafai-lam** [comment]
I found the new release was broken. 

Found that there are a lot of generated code included in the released package. For v0.2.2, those generated code is based on `tests/data/schema.prisma`. However, for this release, codes are partially from `tests/data/schema.prisma` and some from `examples/flask-url-shortener/schema.prisma`. And the Inconsistency break the generator.

I guess it is due to the `make cleanup` command does not run properly before release?

**Reply by RobertCraigie** [reply]
I did run the cleanup command before packaging, that is weird.

Where are you finding the generated code? Navigating to https://pypi.org/project/prisma/#files and downloading the source tar, I can only see non-generated code.

**Reply by RobertCraigie** [reply]
I can confirm this issue, installing prisma in a fresh venv then importing it will break

**Reply by RobertCraigie** [reply]
I have yanked this release

**Reply by RobertCraigie** [reply]
I will fix this issue as soon as possible and re-release it under 0.2.4.

I will also put mechanisms in place to prevent this from happening in the future.

Thank you for reporting the issue

**Reply by kafai-lam** [reply]
Thanks for you quick reply! Wanna report what I meet in detail.

After run `pip install prisma` in clean venv, some extra code generated which is similar as result of running `prisma generate --schema=tests/data/schema.prisma`. Although it is weird, it still work. Since those generated code is consistent, I can overwrite it later with my own schema.

But if I install it in another venv with my daily using package manager `poetry`(`poetry add prisma`), the inconsistency happened.
 
In `models.py`:
```
...
class Url(BaseModel):
    id: int
    created: datetime.datetime
    original: str
    clicks: int
...
```
but meanwhile in `actions.py`:
```
...
    async def find_many(
        self,
        take: Optional[int] = None,
        skip: Optional[int] = None,
        where: Optional[types.PostWhereInput] = None,
        cursor: Optional[types.PostWhereUniqueInput] = None,
        include: Optional[types.PostInclude] = None,
        order: Optional[Union[types.PostOrderByInput, List[types.PostOrderByInput]]] = None,
    ) -> List[BaseModelT]
...
```

Clearly, they come from 2 different schema and I am not able to run any `prisma` command after that.

**Reply by RobertCraigie** [reply]
Oh I see what has happened here, I've been working on the `wip/custom-models` branch and this generates a new file: `prisma/actions.py`. As it is a new file, the code on the `main` branch doesn't know about it and therefore cannot correctly clean it up.

Thank you for investigating and for a very detailed response!
